executor/sort.go

```diff
@@ -47,21 +47,29 @@ type SortExec struct {
 	// keyCmpFuncs is used to compare each ByItem.
 	keyCmpFuncs []chunk.CompareFunc
 	// rowChunks is the chunks to store row values.
-	rowChunks *chunk.List
+	rowChunks *chunk.RowContainer
 	// rowPointer store the chunk index and row index for each row.
 	rowPtrs []chunk.RowPtr
 
 	memTracker  *memory.Tracker
 	diskTracker *disk.Tracker
 
-	// rowChunksInDisk is the chunks to store row values in disk.
-	rowChunksInDisk *chunk.ListInDisk
-	// rowPtrsInDisk store the disk-chunk index and row index for each row.
-	rowPtrsInDisk []chunk.RowPtr
-	// partitionList is the chunks to store row values in disk for partitions.
-	partitionList []*chunk.ListInDisk
-	// partitionRowPtrs store the disk-chunk index and row index for each row for partitions.
+	// partitionList is the chunks to store row values for partitions.
+	partitionList []*chunk.RowContainer
+	// partitionRowPtrs store the sorted RowPtrs for each row for partitions.
 	partitionRowPtrs [][]chunk.RowPtr
+
+	// sortRows is used to maintain a heap.
+	sortRows []chunk.Row
+	// sortRowsIndex store the partition index for each row.
+	sortRowsIndex []int
+	// partitionConsumedRows store the consumed rows num for each partition.
+	partitionConsumedNum []int
```

Comment:
Could these three entries be put into `topNChunkHeapWithIndex`?

And I think `sortRows` is redundant. For better understading for the code reader, we can introduce a new called struct `partitionPointer` embedded in the `topNChunkHeapWithIndex` to indict the rows in heap.

```go
type partitionPointer struct {
    partitionId int
    consumed int
}
```

Reply:
Ok. I will address. 
But I think `sortRows` is need. The rows should be saved in memory, to avoid loading data too many times from disk for comparing row. 
Or I misunderstand your mind ?